src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/NetworkRequestQueue.java

```diff
@@ -0,0 +1,51 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.net;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * Class that represents a custom bounded and concurrent queue.
+ */
+public class NetworkRequestQueue<T> {
+
+  private static final int DEFAULT_MAX_Q_SIZE = 200;
```

Comment:
Do you think instead of making it a program constant, we make it part of rca.conf ? So that changing this does not require a code deployment just a PA restart

Reply:
Made the queue size for the threadpool executor configurable now.